src/Aspire.Hosting/ApplicationModel/ContainerPortReference.cs

```diff
@@ -22,6 +33,27 @@ public class ContainerPortReference(IResource resource) : IManifestExpressionPro
     /// <inheritdoc/>
     public IEnumerable<object> References => [Resource];
 
+    /// <summary>
+    /// Gets the container port value asynchronously.
+    /// </summary>
+    /// <param name="cancellationToken">A cancellation token.</param>
+    /// <returns>
+    /// A task that represents the asynchronous operation. The task result contains the target port 
+    /// from the first endpoint annotation if available, otherwise "8080".
+    /// </returns>
     ValueTask<string?> IValueProvider.GetValueAsync(CancellationToken cancellationToken)
-        => ValueTask.FromResult<string?>("8080");
+    {
+        // If the resource has endpoints defined, try to get the target port from the first endpoint
+        if (Resource is IResourceWithEndpoints resourceWithEndpoints)
```

Comment:
Do we need this cast? Should we just look for annotations, regardless of the .NET type of the Resource?

Reply:
Also needs to filter for http endpoints

Reply:
Why `http` endpoints? Can't a container expose a non-http endpoint?

Reply:
This class is only used for projects AFAIK.